### PR TITLE
FP-2619/FP-2640: Add custom page view tag support for bing

### PR DESCRIPTION
**Pull request by ankurgupta1995**
- Remove old bing pageview tag that did not work for server side events
- fix syntax error

### CODE REVIEW

**Review by ScottsoRobinson — APPROVED** [review]

File: template.tpl [hunk 1]
```diff
@@ -943,10 +943,11 @@ ___TEMPLATE_PARAMETERS___
     "name": "bingEventType",
     "displayName": "Track Type",
     "macrosInSelect": false,
+    "notSetText": "-",
     "selectItems": [
       {
-        "value": "PAGE_LOAD",
-        "displayValue": "page view"
+        "value": "CUSTOM_PAGE_LOAD",
```
[the remark this answers is not in the document]
Comment (ankurgupta1995) [reply template.tpl:949]: That's a good callout. Let me try this out.

[the remark this answers is not in the document]
Comment (ankurgupta1995) [reply template.tpl:949]: I tested this out by setting up the old bing page view tag and then swapping out the template code. It seems like the old tag does not fire a track event, which is the expected and correct behavior here.
TLDR; I think we're fine.

[the remark this answers is not in the document]
Comment (bucrogers) [reply template.tpl:949]: See my latest commit -

Here's what was happening before my commit, for the deprecated PAGE_LOAD:
* We were submitting a track call to the SDK, with eventname === undefined
* Though this didn't result in a track call sent over the wire, as you observed, I didn't think it was clean, and if the SDK were ever to change how it handled undefined event names, could result in unwanted behavior or console messages

After my commit:
* For PAGE_LOAD, we don't submit track to SDK
* Also added a check for non-truthy eventName in the track() function, since we've had reports of errors when eventName was null

**Review by bucrogers — APPROVED** [review]

File: template.tpl [hunk 2]
```diff
@@ -2909,20 +2917,20 @@ const identify = (userID, props, options) => {
   });
 };
 
+// track returns true if valid (truthy) event name, false otherwise
+//  optional for caller to check return value - this is mainly to avoid invalid sdk track calls
 const track = (eventName, props, options) => {
+  if (!eventName) {
+    return false;
+  }
```
Comment (ankurgupta1995) [inline template.tpl:2925]: Nice!

File: template.tpl (hunk 1 above)
[the remark this answers is not in the document]
Comment (ankurgupta1995) [reply template.tpl:949]: Cool! That makes sense. Did you see the event with the undefined name in the network tab or somewhere else? We checked that the ingest worked correctly but did not check the network tab which was a miss on our part.

[the remark this answers is not in the document]
Comment (bucrogers) [reply template.tpl:949]: did not show on network tab (track call not sent over the wire, apparently because passed eventName was undefined).

but it was being submitted to the SDK, which this bypasses.

File: template.tpl [hunk 3]
```diff
@@ -942,11 +942,18 @@ ___TEMPLATE_PARAMETERS___
     "type": "SELECT",
     "name": "bingEventType",
     "displayName": "Track Type",
+    "help": "Select 'Page View' only when your Freshpaint Bing Ads configuration has 'Automatically Send Page View Events' unchecked",
     "macrosInSelect": false,
+    "notSetText": "-",
```
Comment (bucrogers) [inline template.tpl:947]: @ScottsoRobinson @ankurgupta1995 : Added help text here re custom page view usage

File: template.tpl [hunk 4]
```diff
@@ -942,11 +942,18 @@ ___TEMPLATE_PARAMETERS___
...
+    "valueValidators": [
+      {
+        "type": "NON_EMPTY"
+      }
+    ],
```
Comment (bucrogers) [inline template.tpl:952]: @ScottsoRobinson @ankurgupta1995 - added non-empty validator here, which is needed when notSetText is used to ensure an invalid bing tag can't be saved

**Review by bucrogers — APPROVED** [review]

File: template.tpl (hunk 1 above)
[the remark this answers is not in the document]
Comment (ScottsoRobinson) [reply template.tpl:955]: For our understanding for testing in future, what did you do to observe it was submitted to the SDK?

**Review by ScottsoRobinson — APPROVED** [review]

File: template.tpl (hunk 1 above)
[the remark this answers is not in the document]
Comment (bucrogers) [reply template.tpl:955]: two ways:
1. By inspection of the code flow, you could see it through to track
2. I added temporary log() statements to see that it made it thru sdk call
